bl-ap-imgproc/bl_ap_imgproc/bl_ap_imgproc.py

```diff
@@ -0,0 +1,174 @@
+"""IOC Module."""
+
+import os as _os
+import logging as _log
+import signal as _signal
+from threading import Event as _Event
+
+import pcaspy as _pcaspy
+from pcaspy.tools import ServerThread
+
+from siriuspy import util as _util
+
+from .const import Constants
+from .main import App
+
+
+stop_event = _Event()
+
+
+def _stop_now(signum, frame):
+    _log.info('SIGINT received')
+    global stop_event
+    stop_event.set()
+
+
+def _attribute_access_security_group(server, db):
+    for k, v in db.items():
+        if k.endswith(('-RB', '-Sts', '-Cte', '-Mon')):
+            v.update({'asg': 'rbpv'})
+    path_ = _os.path.abspath(_os.path.dirname(__file__))
+    server.initAccessSecurityFile(path_ + '/access_rules.as')
+
+
+class _Driver(_pcaspy.Driver):
+
+    def __init__(self, app):
+        super().__init__()
+        self.app = app
+        self.app.driver = self
+        self.app.init_driver()
+
+    def read(self, reason):
+        _log.debug("Reading {0:s}.".format(reason))
+        return super().read(reason)
+
+    def write(self, reason, value):
+        if not self._isValid(reason, value):
+            return False
+
+        # NOTE: app.write should update driver pv values in database
+        ret_val = self.app.write(reason, value)
+        if ret_val:
+            _log.info('YES Write %s: %s', reason, str(value))
+        else:
+            value = self.getParam(reason)
+            _log.warning('NO Write %s: %s', reason, str(value))
+        return True
+
+    def check_read_only(self, reason):
+        is_read_only = reason.endswith(('-Sts', '-RB', '-Mon', '-Cte'))
+        if is_read_only:
+            _log.debug('PV {0:s} is read only.'.format(reason))
+        return is_read_only
+
+    def check_value_none(self, val):
+        if val is None:
+            msg = 'client tried to set None value. refusing...'
+            _log.error(msg)
+            return True
+        return False
+
+    def check_enums_nok(self, reason, val):
+        enums = self.getParamInfo(reason, info_keys=('enums', ))['enums']
+        if not enums:
+            return False  # not enum
+        elif isinstance(val, int) and val < len(enums):
+            return False  # enum index in range
+        else:
+            return True  # enum index out of range
+
+    def _isValid(self, reason, value):
+        if self.check_read_only(reason):
+            return False
+        elif self.check_value_none(value):
+            return False
+        elif self.check_enums_nok(reason, value):
+            return False
+        else:
+            return True
+
+
+def defineAbortFunction():
```

Comment:
consider using `_` instead of camelCase letters.

Reply:
yes, @fernandohds564 , @RafaelLyra8 is still learning our lib structures with this project. I think we can standardize gradually so as not to introduce a too steep learning curve to be overcome that hinders productivity. but simple suggestions I can implement myself in this PR and can instruct @RafaelLyra8.

Reply:
done